### ISSUE TITLE
Charging slow on Werner

### ISSUE THREAD

**Issue by gestom** [strands-project/scitos_robot#72]
Hi,

since yesterday, Werner's charging rate is very low. The charging current is typically below 2A and is fluctuating wildly. The 'charger_status' field internal_error_flag is  True and 'const_current_mode' is fluctuating between True and False. Battery level was about 77% when I checked.
We recorded the 'battery_status' for a few hours - the rosbag is available here: https://lcas.lincoln.ac.uk/owncloud/index.php/s/FbRBWMZnZQQjVcQ.

Can we do anything to figure out why is the charging slow ? normally it's about 10A.

Best,

Tom

**Comment by marc-hanheide** [comment]
@creuther as our valuable MetraLabs contact, could you forward this to the relevant people, please?

**Comment by creuther** [comment]
@marc-hanheide Of course, thanks for notifying me. Without knowing any details, I think the charger would need repair or even replacement so can you give me a c/o address so we can expedite the whole process a bit and get a new charger sent out? At least that is what I suspect will happen, I can only confirm it tomorrow morning though.

**Comment by marc-hanheide** [comment]
Thanks for coming back to us so quickly. If there's any diagnosis we can do, please let us know.

**Comment by gestom** [comment]
Maybe that the reason for slow charging is caused by the additionally installed PCs inside of the robot, which might cause the robot to overheat. We have never observed the charging slowdown with a robot that had it's skirt off. To test it, I let the other robot (LINDA) patrol 'naked' to deplete its batteries to 20%, put the robot skirt on, started the deployment systems, placed a digital thermometer inside of the robot (at the top part of the skirt area, approximately at the level of the PTU controller) and let the robot charge in a room with controlled 30 °C temperature. First, the robot would charge at 10 A, but as the temperature inside of it would increase, the charging current would drop to +-2A. After the battery was charged to 35%, I removed the skirt and set the room air condition to 19°C. The charging current started to slowly increase again, see the attached graphs.

![drawdepend](https://cloud.githubusercontent.com/assets/4293593/13866895/a5ae4f40-ecb1-11e5-9d9d-c0a208ab88a0.png)
![drawtime](https://cloud.githubusercontent.com/assets/4293593/13866898/ad20d108-ecb1-11e5-8deb-c2410be1da2a.png)

**Comment by marc-hanheide** [comment]
Wow, thanks, guys. This is on Linda, so the question is if the effect is the same on Werner. How did you measure the temperature? Have we got any idea what the temperature profile is on Werner?
Also, have we got a chance to measure the temperature in the battery cells themselves? It would now be interesting to see if we start charging without skirt and at 19℃ and obtain a very different profile...

**Comment by creuther** [comment]
Great analysis, thanks for that! I have created a service ticket and have some diagnostics instructions for you.

First of all, we suggest flashing the latest firmware of the charger as it does improve the stability of the charging "arc", but obviously it won't solve the temperature dependency if that turns out to be the cause. We don't have the best track record with firmwares and usually believe in not changing running systems, but I personally feel very confident about our latest firmware bunch. It includes a couple of fixes, among others it noticeably improves the odometry stability (thanks to the great investigative work by @cburbridge). I will write an email about this to the STRANDS mailing list within the next hour.

As for diagnostics, it would be more helpful for us if you could create a MIRA tape of the /robot/charger/BatteryState channel (topic) while this phenomenon is happening. Doing that is fairly easy, you can just connect to the locally running MIRA instance (it should by default be port 1234) by using
`miracenter -k 127.0.0.1:1234`
and then see whether there is already a "Recorder" tab in the bottom center workspace area. If not, just add the view part by going "Window -> Show View -> Tape -> Recorder". In the widget, press the green "+" button and select the /robot/charger/BatteryState channel. After that you just need to press the red "Record" button and specify a filename, then it'll record the channel. Exactly the same as ROS bags, but readable by our diagnostics tools ;-) Also you need to press "Stop" when done for the tape to finish writing.

There is also a way to access the persistently stored errors of the charger, by using the RPC `/robot/Robot.savePersistentErrors(std::string filename)`
This can be called from miracenter using the RPC Console view (if not already added, add in the same way as above). This persistent error file, if there are any stored, would be helpful as well for the diagnostics.

**Comment by creuther** [comment]
Just in case, here is the file sent to the STRANDS mailing list. 
[scitos_firmware_upgrade_17032016.tar.gz](https://github.com/strands-project/scitos_robot/files/179375/scitos_firmware_upgrade_17032016.tar.gz)

**Comment by gestom** [comment]
How did you measure the temperature? 
Using the Temper1F sensor, see the metallic piece in front of the router.

Have we got any idea what the temperature profile is on Werner?
No. But hospitals might have some (contact less) thermometers and people who can use them. I am not sure it these work on robot skin as well.

Also, have we got a chance to measure the temperature in the battery cells themselves? 
It would now be interesting to see if we start charging without skirt and at 19℃ and obtain a very different profile...
We might try. However, this might not be the case of the batteries only, but the other charging circuitry as well. 
![imag1266](https://cloud.githubusercontent.com/assets/4293593/13876296/e90589e0-ecfa-11e5-88b6-fe53dc8886be.jpg)
![drawtime](https://cloud.githubusercontent.com/assets/4293593/13876304/f65ca3e4-ecfa-11e5-8bda-b5032a7edf02.png)

**Comment by gestom** [comment]
I am about to try the firmware upgrade. However, how do I backup the previous firmware ?

**Comment by marc-hanheide** [comment]
I recall haven't read somewhere "you can't", but I can't find it...

**Comment by gestom** [comment]
so we might end up with a broken system ?

**Comment by gestom** [comment]
I will do some outstanding experiments before then :)

**Comment by gestom** [comment]
OK, I did the firmware and run the robot around with and without the cover. Navigation was working fine and the charging current fluctuations were reduced. The average charging current was slightly higher than before the update, but one could still see that with the cover on, the charging took much longer time.

**Comment by marc-hanheide** [comment]
So, would you then recommend we update Henry? We have an easter break at the moment anyway...

**Comment by gestom** [comment]
Actually, I would not recommend it. Although the charging current is not fluctuating after the update, its average value did not change much and the overheating problem persists. Rather than updating firmware, I recommend upgrading hardware and cut some additional venting hole(s) in the robot shell. I am currently running another batch of tests with the lower skirt removed. This might give us a hint where to cut.

**Comment by gestom** [comment]
So, removal of the lower skirt also helps, so IMHO we might start trying to drill a venting venting hole in the rear part of the robot 'lower' skirt. The question is if Metralabs can provide us with a few replacement plastic parts of the robot's shell and for how much ?

**Comment by creuther** [comment]
I can find out for you, what exactly do you need? Which parts? How many of them? In the same colour as Werner or just some generic colour?

Besides, what I find a bit odd is that the problem arose so suddenly. I can see from your measurements that there is some correlation with temperature, but before it was charging fine despite the exact same situation (shell on) right? If so, then I wouldn't rule out a problem with the charger yet. If you could do the diagnostics I posted above, we might be able to find out more.

**Comment by gestom** [comment]
Maybe the problems are not observed that much because when not used in deployments, the robots were running half-naked.  Moreover, I think that we observed the overheating problems last year as well, but this time it might be more severe. I have plenty of rosbags from the experiments, cannot they substitute the  MIRA tape ? The topic is the same.

**Comment by gestom** [comment]
miracenter -k 127.0.0.1:1234 reports that it 
'could not connect to the framework'

**Comment by creuther** [comment]
I had a look at the scitos_mira node, apparently the MIRA framework there doesn't start with a port open for incoming connections unless you specify the `server_port`variable. But either way, you can also run a separate MIRA instance when the scitos_mira node is **not** running (as only one connection to the CAN bus is allowed).

`miracenter SCITOSConfigs:etc/SCITOS-mapping.xml --variables robot=SCITOS-A5,canType=MLCAN`

The reason we cannot easily use the ROS bags instead is that we have tools that read the tape files (and channels within) for analysis. I guess it would be possible to build a converter but as this can't be done generically, we have never put too much priority into it.

**Comment by gestom** [comment]
OK, the tape recorded by @Jailander is at labe.felk.cvut.cz/~tkrajnik/miratapebattery.tape. The persistent errors of the charger are  at labe.felk.cvut.cz/~tkrajnik/persistent.errors. We will probably need one 'bottom' skirt and one 'back-middle' skirt with Linda-blue color.

**Comment by creuther** [comment]
Thanks, we are analyzing the tape and error files and will get back to you. As for the case parts / skirts, I talked to my boss about it. 
Unfortunately we can only sell the large middle parts as a whole / together as they are commissioned in that way. So the two middle parts and the bottom part including paintwork would cost 750€ net altogether.  Let me know whether you want a formal quote or how you want to continue. Otherwise I'll be back in touch about the charging irregularities.

**Comment by marc-hanheide** [comment]
OK, as this tape was on Linda, we shall create one from Werner as well.

**Comment by cdondrup** [comment]
I recorded a bit of data:
https://lcas.lincoln.ac.uk/owncloud/index.php/s/VwGbXIiOWIBqhk0

This contains 4 files:
- persistenterrors.txt which is the persistent error log
- a rosbag file which contains recordings of the charging topic while the robot docks and undocks
- `charger.tape` a mira tape of the robot being on the charger
- `drop.tape` a mira tape that hopefully shows the drop in current when docking. This was created by sending the robot to the charger and when it docked, immediately killing the scitos node and starting mira centre. It does not have the docking in it but should start with the high current we observed and then drop.

**Comment by marc-hanheide** [comment]
Here is the plot of the rosbag @cdondrup created:
![image](https://cloud.githubusercontent.com/assets/1153084/14327605/54d6c5b6-fc33-11e5-9744-239719c9dd82.png)

Sorry, about the x axes... it's samples, so roughly it's second \* 10.

I plotted the current (as before), and we see a similar behaviour. indeed we went through a number of docking and undocking cycles, where we always observed pretty much the same behaviour. After docking the charging started at a high rate (so negative current, red line, at about -8A, one such docking starts at 2362 in the plot). At that point we also see the Voltage (green line) to steeply go up, but interestingly slowly, and not directly with the current. At 3537 we see the dropping occurring when the charger seems to go give less power to the robot, until levelling out at about -3A from 4524 onwards. We have then repeated this procedure three times, quickly undocking and re-docking to see if it makes any difference in the speed it declines. One hypothesis was that if we charged before, the unit might still be warm and the decline in charging speed should be faster than at the beginning when the robot was away from the docking station for longer... this seems to be the case as indeed in the last cycle the "good" charging duration was only about half long as in he first cycle.

_Note:_ This time it continued charging because we had a very low load on the robot, not runnng anything but the bare necessary things to run the system. On full load, those 3A would be eaten by the computer running stuff. That's what we currently see: The current is just about enough to power the processing, but not to charge. It would be perfectly enough if the higher charge we see at the beginning would be maintained.

**Comment by marc-hanheide** [comment]
So, @creuther is this helpful? @gestom, I think this all confirms what we were suspecting, right?

**Comment by marc-hanheide** [comment]
One last note... the voltage profile is a little surprising to me, which is why I also included it, maybe it help diagnosis.

**Comment by creuther** [comment]
@marc-hanheide @TobKoer We sent you a new charger today, I'll provide you with instructions on how to change it soon. 

The additional information you provided is quite interesting as well, I forwarded it to our hardware guys. I'll keep you posted!

**Comment by marc-hanheide** [comment]
I forgot to add, that this was all measured when the robot was about 50% charged, so to make sure we don't end up in the kind of tickle charge when it is nearly full...

**Comment by cdondrup** [comment]
"tickle charge" hihi XD

**Comment by marc-hanheide** [comment]
@cdondrup also happy to amuse you.

@creuther are you then confident that the profiles shown result from a faulty charging unit? In other words, shall we be optimistic that a new charger will resolve the situation? Or should we consider further alternatives?

**Comment by creuther** [comment]
@marc-hanheide Somewhat confident but I agree that the behaviour observed is kind of odd. Waiting for an expert analysis of the data but from what I see, the robot experienced cell undervoltage quite a few times. It does not look like any of the battery cells dropped below the critical threshold of 2.0V (after which they are considered permanently damaged by the manufacturer of the cells), so they should be okay.

**Comment by marc-hanheide** [comment]
@creuther the charger has arrive, but we now urgently need the instructions on how to fit it.

**Comment by marc-hanheide** [comment]
So, new charger, old story: When initially starting the charge it started with about 5-6A charging, not the 8A we saw previously, but that was a "bug" anyway, as the charger should be limited to a total of 10A according to @creuther.

Then,  we saw it slowly drop as temperature likely got up. After this initial drop, I recorded the whole profile overnight:
![bat](https://cloud.githubusercontent.com/assets/1153084/14457481/0b9c5448-00ab-11e6-8b5d-0400fd9b1e35.png)

The X axis shows hours, with 0 being midnight local time. So, we can see that henry undocked as intended shortly after 9am when the charging finished, but from this plot I can also guess that @cdondrup probably started logging at 8:20, as that’s where the charging goes down to 0 again. 
In short, it seems that we didn't win anything. 

At the beginning of this plot, `/charger_status` showed this:

```
---
header:
  seq: 63144
  stamp:
    secs: 1460400047
    nsecs: 169556000
  frame_id: ''
charging: True
empty: False
full: False
derating: True
charging_disabled: False
const_volt_mode: False
const_current_mode: False
internal_error_flag: False
```

The `derating: True` was suspicious and @creuther said it might indicate that the charger has reduced its power due to heat. This is to be confirmed.

**Comment by marc-hanheide** [comment]
So, we can also deduce that we loose about 10% per hour, and we gain (without logging, with it we seem to gain nothing!) max 5% per hour on charging.

**Comment by marc-hanheide** [comment]
Given our experience with both Werner and Linda I strongly advise the TSC deployment people to look at and consider this. We have seen little autonomous behaviour at aaf due to power problems. I suggest you do some tests with your robot under full load early to assess the charging profiles. @hawesie (at whoever is in charge)

**Comment by hawesie** [comment]
Oh poop, this is the first I've seen on this. @bfalacerda and you work with @creuther to see what happens with Betty? Would it help if we try to fit a cooling fan into the skirt somehow?

**Comment by Jailander** [comment]
@hawesie we are currently fitting a fan in Linda's lower skirt will send you a pic as soon as we get the part back.

**Comment by Jailander** [comment]
Also maybe its worth taking a look at [Marc's Awesome launch action server](https://github.com/strands-project/strands_apps/tree/indigo-devel/roslaunch_axserver) to launch and stop components that are not necessary whilst charging.

**Comment by marc-hanheide** [comment]
The Problem is that the biggest drainer is logging for us. If we stop that any time we offer services at the charging station we have quite patchy data, but it seems we have to do that.

**Comment by agcohn** [comment]
and/or cooling holes drilled in skirt?

Tony

## 

Prof A G Cohn, FREng, CEng, CITP, FAAAI, FECCAI, FAISB, FIET, FBCS
School of Computing
University of Leeds
Leeds LS2 9JT
0113 3435482     www.comp.leeds.ac.uk/agc

**Comment by marc-hanheide** [comment]
Yes, these options are all under consideration and being assessed. But we'd live advice from @creuther and metralabs.

**Comment by bfalacerda** [comment]
yeah, I've been kind of following what's going on. @cburbridge 's suggestion before leaving was to add a fan like this:

![img_2530](https://cloud.githubusercontent.com/assets/4320588/14460266/8877a4b4-00b4-11e6-83b3-7240777fd4a4.JPG)

We'll do it next week when  @maximeadjigble is back. I also think it might be a good idea to avoid using the second pc if possible, according to @creuther they are using quite a bit of power, and with both on we'll probably struggle to achieve full charges during night regardless of heating issues.

**Comment by marc-hanheide** [comment]
We considered taking the second one out, but the effects has been deemed minimal aus they use very little power when not under load (we have seen). Modern PCs are pretty good at power management, so having them there when they are not computing is almost as not having them, And hence, two PCs at 30% load end up as pretty much the same as one under 60% load. We have seen this also, when we don;t run anything, but the PCs idle, we get plenty of charge, so it is a load problem after all, and a heat problem.

**Comment by marc-hanheide** [comment]
The fan at that position is unlikely to cool the charger (which sits in the lower skirt above the wheel) effectively.

**Comment by marc-hanheide** [comment]
We may want to also add another heat sink to the charger, but again, best to hear advice from @creuther.

**Comment by Jailander** [comment]
Also I'm not sure you'll get much cooling from putting the fan there, unless, you have an escape on top of that skirt you will end up moving hot air only (which I admit still cools down but not as much), and not exchanging air with the outside of the robot.

**Comment by cdondrup** [comment]
> So, we can see that henry undocked as intended shortly after 9am when the charging finished, but from this plot I can also guess that @cdondrup probably started logging at 8:20, as that’s where the charging goes down to 0 again.

I started the logging at around 8  while still on the charging station. So I guess you're right, that is what we see in the plot.

**Comment by bfalacerda** [comment]
The idea was to make the air flow up, there are some wholes when the skirt is on, and also on the neck and stuff

**Comment by marc-hanheide** [comment]
But @bfalacerda the charger is below the plate, so there won't be any circulation there.

**Comment by bfalacerda** [comment]
yeah, I see that. Just to confirm, these problems also happen with the skirt off?

**Comment by Jailander** [comment]
yes that is why we are drilling the hole on the lower skirt will go in a bit to get you a picture

**Comment by cdondrup** [comment]
The problems do not seem to happen with the skirt off, no.

**Comment by cdondrup** [comment]
OK, just to add to that. There still is an effect due to heat but the effect is smaller with the skirt off. So the charging is still reduce but not as much.

**Comment by marc-hanheide** [comment]
Position of the charger:
![image](https://cloud.githubusercontent.com/assets/1153084/14460615/bde8a60c-00be-11e6-932c-99ed86053d28.png)

**Comment by marc-hanheide** [comment]
So, I guess if we improve the conduction of heat to the plate where the charger is mounted and then cool the air above it, it might have a good effect... Anyone every tried if the plate above the charger is warm/hot to the touch?

**Comment by bfalacerda** [comment]
ok, so if we manage to reduce the temperature inside the pc area when the skirt is on, we might not need to go to the bottom part? Anyway, if you guys find a solution this week we'll follow suit

**Comment by marc-hanheide** [comment]
It probably doesn't help that the GPU of the side PC is blowing its hot air right at the place where the charger sits...

**Comment by cdondrup** [comment]
Not using the GPU for anything currently. Can't be that hot...

**Comment by marc-hanheide** [comment]
Then, it might even have a cooling effect???

**Comment by marc-hanheide** [comment]
Anyway, can you guys check with Linda if the charger actually gets hot?

**Comment by Jailander** [comment]
Yes the plate gets hot, with and without the skirt I imagine the effect is reduced when the upper skirt is off because of the improved heat transfer. but the plate is always moderately warm

**Comment by Jailander** [comment]
I'll check in a second, we need a thermal camera ;)

**Comment by Jailander** [comment]
Ok Linda has a full battery right now so the charger is cold, I'll let her discharge for an hour and plug her and see after a while, however since the lower skirt is not here it won't be as warm as in other cases.

**Comment by marc-hanheide** [comment]
We have a thermal camera, ask Greg.

**Comment by Jailander** [comment]
haha I know, just realised we could have used it before I made the comment, but right now it doesn't make sense since we don't have the skirt. But I'll ask Greg for it when we have the part back.

**Comment by creuther** [comment]
I have been doing some asking around and had a look at Bob but it seems while I was doing this there were quite a few new posts. Anyway, I'll just continue with what I started typing before ;)

@marc-hanheide Unfortunately the life-percentage is not really that reliable, so it would be more helpful to see voltage plotted against current. What kind of services were running during the night, and what kind of power draw do you except the robot to have due to them? If the robot was still drawing around 6A then the numbers seen might make sense.
Also, the charger per se should not need excessive cooling. The only time we have run into a temperature problem was with the HOBBIT project, which had a quite similar setup in terms of complexity and power consumption. The reasons there were heat build-up due to insufficient air circulation as well. So as @cdondrup also confirms that the problems do not happen with the skirt off, we have to improve the circulation / get rid of the heat build-up.

In terms of the derating flag, I have some pleasant news to deliver - there is yet another bug in the firmware at the moment that automatically sets the flag to true when the robot is charging. But heat will still most likely be the issue.

To see whether everything else in terms of charging is working correctly, I'd suggest again letting the robot charge with all computers shut down and then reading the current and voltage from the status LED display. It should read around -10A.

Bear in mind though that as soon the battery is full (voltage around 29V +-0.5V), the current will drop to 0 and then hover around small positive and negative currents.

But all in all it seems most likely that everything is in order and that the heat is the problem at the moment.

**Comment by Jailander** [comment]
@creuther in one of those charts @gestom made when we recorded the tape you could tell that the problem also happens with the skirt off just in a much lower scale, however, this could be happening on Linda because of the defective cell. Anyway solving the heat build up issue seems to improve charging.

**Comment by marc-hanheide** [comment]
Thanks @creuther, so given all this, have your MetraLabs guys any advice how to best get rid of the heat build up? What did you do in Hobbit? 
Taking the skirt off is no option. I agree that what we see now is mostly consistent, and when it was cool, we saw indeed that `10 A - current_drawn = current_charge`. 

So, the charger is functioning as it should, the firmware apparently isn't? Can we set the derated to false somehow?

**Comment by creuther** [comment]
@marc-hanheide I'm waiting for their response on it. The firmware bug is only a "display" issue, the charger isn't necessarily going into derating mode but when it is on the charging, the derating flag is being shown as set regardless of the internal state.

Apparently it is possible to see the actual power draw of the charger when on the charging station, but this data is not yet exposed in the charger status channel / topic (it is certainly the first I've heard of it). It is nevertheless present in one of the CAN bus PDOs and should be viewable with moderate effort. I am testing it on Bob now, I'll report back when I'm done.

**Comment by bfalacerda** [comment]
can someone point me to the scripts to do the nice plots, so we leave betty charging tonight?

**Comment by marc-hanheide** [comment]
I don't have a script. I used
1. `rosrun topic_tools throle messages /battery_state 1 /battery_state_throttled` to limit the data recorded to once per second
2. `rostopic echo -p /battery_state_throttled  > batt.csv` used to record this into a CSV file directly
3. downloaded this file to my laptop
4. `grep -v "%" batt.csv | tr "," "\t" > batt.dat` to convert it into something gnuplot understands
5. run `gnuplot` with the following command: 
   
   ```
   plot "bat.dat" using (($3-1460397600000000000)/3600000000000-4):5 axes x1y1 title "Current [A]", "bat.dat" using (($3-1460397600000000000)/3600000000000-4):6 axes x1y2 title "lifetime" with lines
   ```
   
   where 1460397600000000000 is the [UNIX epoch](http://www.epochconverter.com/) multiplied by 10e9 to get the starting time of the plot... and 3600000000000 is how long an hour is in nano seconds ;-)

Feel free to cast into a script!

**Comment by creuther** [comment]
So I tested the PDO stuff on Bob and it seems to work, even though it not that user-friendly. You have to get exclusive access to the CAN bus (i.e. every other MIRA instance accessing the CAN bus has to be killed) and then opening miracenter e.g. like this: `miracenter SCITOSConfigs:etc/SCITOS-mapping.xml --variables robot=SCITOS-A5,canType=MLCAN`

Then open the "CAN Tools View" by going  "Window -> Show view -> CAN Tools View". You should see something like this:
![cantoolsview_pdo184](https://cloud.githubusercontent.com/assets/7942709/14463364/a291c0d8-00c2-11e6-9170-1e23d7ffe8b8.jpg)

PDO 0x183 and 0x184 are the interesting ones. Bytes 3 and 2 will indicate the voltage in mV and bytes 5 and 4 will indicate the current in mA. 0x183 is the battery voltage and current, and 0x184 is the charging station (ACS) voltage and current. So for example for PDO 0x184 we get `voltage = 89ef hex = 35.3 dec` (least signifcant byte first or whatever, anyway the second one needs to go first).

To get the overall current provided by the charger, you do
`I_charger = (V_acs * I_acs / V_battery) * 0.95`
And of course it holds that `I_charger = I_draw + I_battery` so we can see the power draw of the robot and its components as well.

A note on my part: It would be easier for these diagnostics to adapt the server_port parameter [here](https://github.com/strands-project/scitos_drivers/blob/indigo-devel/scitos_mira/src/scitos_node.cpp#L54) to enable interprocess communication between MIRA frameworks. This way, we wouldn't have to kill anything to access MIRA stuff, we could just connect to the running instance by doing `miracenter -k ip:port`.

**Comment by bfalacerda** [comment]
So we got betty to use ~8A by loading all her processors, and then put her to charge with everything closed. Looks good I think:
![drawing](https://cloud.githubusercontent.com/assets/4320588/14491611/d1c7260e-0171-11e6-8efd-a83d0a4cdf37.png)

**Comment by bfalacerda** [comment]
Just to clarify,battery starts dropping once @ferdianjovan started doing stuff with her

**Comment by marc-hanheide** [comment]
Wow, this looks nice... So, @creuther what is the difference between this newer robot (betty is the new one!) and the older versions? @bfalacerda can you do the same with bob, just to see if there is a difference?

**Comment by bfalacerda** [comment]
Bob has a broken charger thingy, so he only charges at 5A tops (we're getting a new one). I left him charging with a cable (we only have 1 charging station in Bham), and with skirt off, and this is what it looks like. It's less stable, but it also looks good I think. I'll try the same with the skirt on, but atm if we use more than 5A he'll be losing battery even when plugged in. That makes it harder to heat up the system... How should I do it?
![drawing](https://cloud.githubusercontent.com/assets/4320588/14492304/5ca00d64-0176-11e6-84d7-f54fed906ef0.png)

**Comment by creuther** [comment]
@marc-hanheide Especially with a new charger, there should be no significant difference between the robots. As mentioned before, can you run another one of these plots with voltage instead of lifetime? Especially after changing the charger but also in general, the lifetime is (unfortunately) not the most reliable metric.
In your last plot it _could_ in theory be that the robot reached full charge at around 8am, though I know that it is unlikely. We would be able to see that by plotting the voltage, which would level out when the battery is fully charged (and then jittering as charging is turned on and off periodically to keep the robot at full charge but not overcharge it).

Also, to exclude any issues we should really do a test where the robot is charging from a non-full state with all computers and external peripherals turned off. The status LED will still show the voltage and current, and there we could see how well the robot charges without any kind of load.

**Comment by gestom** [comment]
Hi, since Betty had some problems with her charger at TSC, we left Linda's charger there and took Betty's to Lincoln. With Linda, the charging is working quite OK, but we still had the overheating problem. When charging, Linda's charging current initially started about 6A, but it dropped quickly to 3A, which makes charging rather slow. We went for a hardware solution with a 92 mm fan, see
![fan](https://cloud.githubusercontent.com/assets/4293593/16389228/e81ec214-3c94-11e6-84b6-7b68867ad054.jpg).

We tried three full charging cycles: without the vent, vent only and vent+fan. The results are attached as well.

![charging](https://cloud.githubusercontent.com/assets/4293593/16389301/2b4f6e94-3c95-11e6-9509-41a4d1d69205.png)

So for us, the HW solution to prevent the overheat seems to work quite OK.

**Comment by marc-hanheide** [comment]
<3

**Comment by cdondrup** [comment]
Why is this assigned to me? Maybe you meant @creuther ?!

**Comment by marc-hanheide** [comment]
I assume so, yes.

**Comment by hawesie** [comment]
@gestom @marc-hanheide do you have instructions somewhere on how to do this or a description of what you did? I'd like to get Betty equipped with a fan before she goes back to TSC.

**Comment by gestom** [comment]
we (our technician) cut a circular vent in the bottom skirt (see the pic) and installed a 92mm 12V PC fan, which we connected to one of the side PCs

**Comment by marc-hanheide** [comment]
@kirumang this is what needs to be done for henry!

**Comment by gestom** [comment]
@kirumang I checked the logs from the last two days and Werner is charging only by 2 Amps. There are two differences to our robot: 1) the fan is 80mm instead of 92 2) the firmware was not probably updated. I will get a silent 92mm fan and update the firmware now. It seems that I will need to come here next week as well.

**Comment by marc-hanheide** [comment]
Boooh... Not again or still...
